tools/docker-network/tests/dockerframework.go

```diff
@@ -368,435 +352,304 @@ func (d *DockerTestFramework) StopIssueCandidacyPayload(nodes ...*Node) {
-			Input:        fundsUTXOOutput,
-		}).
-		AddInput(&builder.TxInput{
-			UnlockTarget: from.AccountOutput.UnlockConditionSet().Address().Address,
-			InputID:      from.OutputID,
-			Input:        from.AccountOutput,
-		}).
-		AddOutput(accTransitionOutput).
-		AddOutput(foundryOutput).
-		SetCreationSlot(currentSlot).
-		AddBlockIssuanceCreditInput(&iotago.BlockIssuanceCreditInput{AccountID: from.AccountID}).
-		AddCommitmentInput(&iotago.CommitmentInput{CommitmentID: lo.Return1(issuerResp.LatestCommitment.ID())}).
-		AllotAllMana(currentSlot, from.AccountID, 0).
-		Build()
+	congestionResp, err := clt.Congestion(ctx, issuerAddress, 0, lo.PanicOnErr(issuerResp.LatestCommitment.ID()))
 	require.NoError(d.Testing, err)
 
-	blkID := d.SubmitPayload(ctx, signedTx, wallet.NewEd25519Account(from.AccountID, from.BlockIssuerKey), congestionResp, issuerResp)
-
-	updatedAccount = &Account{
-		AccountID:      from.AccountID,
-		AccountAddress: from.AccountAddress,
-		BlockIssuerKey: from.BlockIssuerKey,
-		AccountOutput:  accTransitionOutput,
-		OutputID:       iotago.OutputIDFromTransactionIDAndIndex(lo.PanicOnErr(signedTx.Transaction.ID()), 0),
+	if skipCheck {
+		return issuerResp, congestionResp
 	}
 
-	d.AwaitTransactionPayloadAccepted(ctx, blkID)
-
-	fmt.Println("Create native tokens transaction sent, blkID:", blkID.ToHex(), ", txID:", lo.PanicOnErr(signedTx.Transaction.ID()).ToHex(), ", slot:", blkID.Slot())
-
-	// wait for the account to be committed
-	d.AwaitCommitment(blkID.Slot())
-
-	d.AssertIndexerAccount(updatedAccount)
-	d.AssertIndexerFoundry(foundryID)
-
-	return updatedAccount
-}
-
-func (d *DockerTestFramework) CheckAccountStatus(ctx context.Context, blkID iotago.BlockID, txID iotago.TransactionID, creationOutputID iotago.OutputID, accountAddress *iotago.AccountAddress, checkIndexer ...bool) {
-	// request by blockID if provided, otherwise use txID
-	// we take the slot from the blockID in case the tx is created earlier than the block.
-	clt := d.Node("V1").Client
-	slot := blkID.Slot()
+	if congestionResp.BlockIssuanceCredits <= 500 {
+		d.IncreaseBIC(ctx, issuerAddress.AccountID())
 
-	if blkID == iotago.EmptyBlockID {
-		blkMetadata, err := clt.TransactionIncludedBlockMetadata(ctx, txID)
+		// need to renew BlockIssuance and Congestion response bc we waited until BIC is updated (until the allotment is committed)
+		issuerResp, err = clt.BlockIssuance(ctx)
 		require.NoError(d.Testing, err)
 
-		blkID = blkMetadata.BlockID
-		slot = blkMetadata.BlockID.Slot()
+		congestionResp, err = clt.Congestion(ctx, issuerAddress, 0, lo.PanicOnErr(issuerResp.LatestCommitment.ID()))
+		require.NoError(d.Testing, err)
 	}
 
-	d.AwaitTransactionPayloadAccepted(ctx, blkID)
+	return issuerResp, congestionResp
+}
 
-	// wait for the account to be committed
-	d.AwaitCommitment(slot)
+// IncreaseBIC requests faucet funds then uses it to allots mana to an account.
+func (d *DockerTestFramework) IncreaseBIC(ctx context.Context, toId iotago.AccountID) {
```

Comment:
I think we should remove this function as it is not the way a user would/should increase their BIC. We should use the transaction builder utilities which allow additional Mana to be allotted as needed for the block requirements.